GitCommands/RevisionReader.cs

```diff
@@ -172,44 +172,46 @@ public sealed class RevisionReader : IDisposable
             string revisionFilter,
             string pathFilter)
         {
-            return new GitArgumentBuilder("log")
+            var builder = new GitArgumentBuilder("log");
+            builder.Add("-z");
+            builder.Add($"--pretty=format:\"{FullFormat}\"");
+            if (!string.IsNullOrWhiteSpace(branchFilter) && isBranchFilterSimple(branchFilter))
             {
-                "-z",
-                branchFilter,
-                $"--pretty=format:\"{FullFormat}\"",
+                builder.Add(branchFilter);
```

Comment:
--branches accepts wildcards. Is it not sufficient to remove the check for special chars?

Reply:
--branches option [has a weird behavior](https://www.git-scm.com/docs/git-log#Documentation/git-log.txt---branchesltpatterngt), when no wildcards are present in the branchFilter:

>  If pattern lacks ?, *, or [, /* at the end is implied

This is why we have to check if these characters are in branch filter, and where they are not, we use `git log branchFilter` instead